src/scripts/main.js

```diff
@@ -1,3 +1,31 @@
 'use strict';
 
-// write code here
+const parentElement = document.querySelector('ul');
+const employees = Array.from(document.querySelectorAll('li'));
+
+function getSalary(employee) {
+  const salaryString = employee.getAttribute('data-salary');
+  const cleanSalary = salaryString.replace(/[^0-9.-]+/g, '');
+
+  return parseInt(cleanSalary, 10);
+}
+
+function sortList() {
+  employees.sort((a, b) => getSalary(b) - getSalary(a));
+
+  parentElement.innerHTML = '';
+  employees.forEach((employee) => parentElement.appendChild(employee));
+}
+
+function getEmployees() {
+  return employees.map((employee) => ({
+    name: employee.textContent.trim(),
+    salary: getSalary(employee),
+  }));
+}
+
+sortList();
```

Comment:
The 'sortList' function is called immediately, which means the list is sorted as soon as the script runs. Ensure that this is the desired behavior, especially if the list should be sorted based on user interaction or another event.